hi-ml-azure/testazure/testazure/test_azure_util.py

```diff
@@ -2151,79 +2166,99 @@ def _assert_dataframe_properties(df: pd.DataFrame, num_crossval_splits: int) ->
         metrics_df = util.aggregate_hyperdrive_metrics(
             run_id="run_id_123",
             child_run_arg_name="child_run_index",
-            aml_workspace=ws
+            aml_workspace=DEFAULT_WORKSPACE.workspace
         )
         _assert_dataframe_properties(metrics_df, num_crossval_splits)
 
     # test the case where a run object is provided
     metrics_df_2 = util.aggregate_hyperdrive_metrics(
         run=dummy_hyperdrive_run,
         child_run_arg_name="child_run_index",
-        aml_workspace=ws
     )
     _assert_dataframe_properties(metrics_df_2, num_crossval_splits)
 
     # if neither a run or a run_id is passed, an error should be raised
-    with pytest.raises(AssertionError, match="Either run or run_id must be provided"):
-        util.aggregate_hyperdrive_metrics(child_run_arg_name="child_run_index", aml_workspace=ws)
+    with pytest.raises(ValueError, match="Either run or run_id must be provided"):
+        util.aggregate_hyperdrive_metrics(child_run_arg_name="child_run_index")
 
-    # test case where provide an invalid child run arg name
+    # test case with a non-existent metric name
     invalid_metrics = ["test/accuracy", "idontexist"]
     metrics_df_3 = util.aggregate_hyperdrive_metrics(
         run=dummy_hyperdrive_run,
         child_run_arg_name="child_run_index",
         keep_metrics=invalid_metrics,
-        aml_workspace=ws
     )
-    assert len(metrics_df_3) == 1
-    assert list(metrics_df_3.index) == ['test/accuracy']
-
-    # if a workspace or config file isn't provided, an error should not be raised
-    metrics_df_4 = util.aggregate_hyperdrive_metrics(run=dummy_hyperdrive_run, child_run_arg_name="child_run_index")
-    _assert_dataframe_properties(metrics_df_4, num_crossval_splits)
+    assert len(metrics_df_3) == 2
```

Comment:
Is metrics_df_3 a dataframe with an empty list for each of the indices listed in invalid metrics? If so could we make that more explicit here, e.g. `assert len(metrics_df_3) == len(invalid_metrics)` and `assert metrics_df_3.loc[invaid_metrics[0]] == []`? Also we could check that the column names match the child_run_index as expected

Reply:
done